shared/actions/chat/thread-content.js

```diff
@@ -744,24 +744,21 @@ function* _updateThread({
   }
 }
 
-function* _appendMessagesToConversation({
-  payload: {conversationIDKey, messages},
-}: ChatGen.AppendMessagesPayload) {
-  const currentMessages = yield Saga.select(Constants.getConversationMessages, conversationIDKey)
-  const nextMessages = currentMessages.concat(messages.map(m => m.key))
-  yield Saga.put(
-    EntityCreators.replaceEntity(['conversationMessages'], I.Map({[conversationIDKey]: nextMessages}))
-  )
-}
-
-function* _prependMessagesToConversation({
-  payload: {conversationIDKey, messages},
-}: ChatGen.AppendMessagesPayload) {
-  const currentMessages = yield Saga.select(Constants.getConversationMessages, conversationIDKey)
-  const nextMessages = I.OrderedSet(messages.map(m => m.key)).concat(currentMessages)
-  yield Saga.put(
-    EntityCreators.replaceEntity(['conversationMessages'], I.Map({[conversationIDKey]: nextMessages}))
-  )
+function* _appendOrPrependMessagesToConversation(arg: ChatGen.AppendMessagesPayload) {
+  const {conversationIDKey, messages} = arg.payload
+  if (arg.type === ChatGen.appendMessages) {
+    const currentMessages = yield Saga.select(Constants.getConversationMessages, conversationIDKey)
```

Comment:
really currentMessages and the Saga.put are the same so really your if just has to be the const nextMessages lines